src/time_zone_impl.cc

```diff
@@ -58,7 +59,7 @@ bool time_zone::Impl::LoadTimeZone(const std::string& name, time_zone* tz) {
 
   // Check whether the time zone has already been loaded.
   {
-    std::lock_guard<std::mutex> lock(TimeZoneMutex());
+    std::shared_lock<std::shared_mutex> lock(TimeZoneMutex());
```

Comment:
The problem is that our functions will take an optional timezone parameter (string) from the user input. We need to have a mechanism to load it once and store it somewhere (we probably want to avoid using lock again on a map). I am thinking to have a static map <name, timezone> to pre-load all time zones which is not ideal and seems to waste space since most timezones are not used.

Reply:
Thanks a lot for suggestions. I will think a bit more.